test/SparqlAntlrParserTest.cpp

```diff
@@ -855,3 +855,46 @@ TEST(SparqlParser, GroupClause) {
                              IsExpressionGroupKey("COUNT(?baz)"));
   }
 }
+
+TEST(SparqlParser, ValuesClause) {
+  expectCompleteParse(
+      parseValuesClause("VALUES ?test { \"foo\" }", {}),
+      IsValues<vector<std::string>, vector<vector<std::string>>>(
+          {"?test"}, {{"\"foo\""}}));
+  // These are not implemented yet in dataBlockValue
+  // (numericLiteral/booleanLiteral)
+  EXPECT_THROW(parseValuesClause("VALUES ?test { true }", {}), ParseException);
+  EXPECT_THROW(parseValuesClause("VALUES ?test { 10.0 }", {}), ParseException);
```

Comment:
You can also derive a lambda `expectValuesParsingFails` for the failing test cases, that only takes the input. The rest is always redundantly the same, and the `EXPECT_THROW` clauses currently don't fit in one line.